### ISSUE TITLE
Implement MAG dereplication

### ISSUE THREAD

**Issue by misialq** [bokulich-lab/q2-moshpit#4]
We need an action supporting dereplication of MAGs into unique genomes.

**Acceptance criteria:**

- uses dRep ([link](https://drep.readthedocs.io/en/latest/index.html))
- uses `SampleData[MAGs]` + `SampleData[MultiAlignmentMap]` as inputs
- outputs `FeatureData[MAG]` + `FeatureTable[Frequency]`

**Comment by nbokulich** [comment]
That solution works for collapsing by taxonomy, but not dereplication of unique MAGs, as it assumes that all MAGs that classify to the same taxon would necessarily be replicates. So we need a different solution for strain-level profiling.

**Comment by misialq** [comment]
I've been trying to install the `dRep` tool mentioned in the issue description, however it seems like using it on macOS is a bit more tricky - while it does install through conda, it throws an error when trying to run it saying something about outdated version of numpy. It seems that for whatever reason (I gave up now trying to figure that out tbh) the issue arises due to a _very_ old version of scikit-learn being installed in this environment (0.22.*, which uses deprecated numpy methods) due to some other dependency (openmp). It seems impossible to update it to the newer one too (in fact, dRep itself doesn't pin any dependencies). It seems to work ok on Linux 😭  I have not tried pip installation, but I guess we don't care since we want to use conda anyways. I also saw that one can use [DAS_Tool](https://github.com/cmks/DAS_Tool) for the same purpose so I'll look into this one, unless anyone knows any other tool from the top of their head?
